src/watcher.ts

```diff
@@ -1,39 +1,61 @@
 import chokidar from 'chokidar';
 import { exec, ChildProcess } from 'child_process';
+import type Serverless from 'serverless';
 
-export type Config = {
+type ConfigItem = {
   path: string | string[];
-  command: string;
-}[];
+  command?: string;
+  hooks?: string | string[];
+};
+
+export type Config = ConfigItem[];
 
 export interface Watcher {
   start: () => void;
   stop: () => Promise<void>;
 }
 
-export const makeWatcher = (config: Config): Watcher => {
+export const makeWatcher = (config: Config, serverless: Serverless): Watcher => {
   const internalWatchers: chokidar.FSWatcher[] = [];
   const processes: { [pid: number]: ChildProcess } = {};
 
+  const runCommand = (command: string, eventType: string, eventPath: string) => new Promise<void>((resolve) => {
+    const env = {
+      ...process.env,
+      WATCHER_EVENT_TYPE: eventType,
```

Comment:
Note: Was thinking about whether some people would ever want to disable us adding these environment variables. I think it's very unlikely (see other comment about whether this should mean we release a new breaking change version) this bothers anyone, and if it does either they can override it in their command or we can add a config option to disable this in future.